synapse/lib/aha.py

```diff
@@ -1041,6 +1042,11 @@ async def getAhaSvcProxy(self, svcdef, timeout=None):
         if client is None:
             return None
 
+        try:
+            await asyncio.wait_for(client.ready.wait(), timeout=5.0)
```

Comment:
Having a hard coded timeout here bypasses the user specified timeout.

Reply:
Since the `proxy()` method on the Client object already awaits ready status, why is this additional check necessary?

Reply:
This came up when a service was completely offline (unreachable) and the code was waiting indefinitely because the default timeout is none. I figured this is one of those edges cases which is not really tied to the param of the invoking call because we are trying to get a client back within the function and hopefully from a service online which doesn't hold-up everything.

Reply:
If the caller wants the call to not wait indefinitely, they are required to specify a timeout. :)

Reply:
Ok, I can drop it. I do worry if caller isn't aware a service is down it may appear as if things are "stuck".